internal/provider/transfer_resource.go

```diff
@@ -335,3 +301,523 @@ func transferTypeValidator() validator.String {
+			m.Transformation = new(transferTransformation)
+		}
+		diags.Append(m.Transformation.Read(t.Transformation)...)
+	} else {
+		m.Transformation = nil
+	}
+
+	return nil
+}
+
+func (m *transferResourceModel) UpdateRequest() (*transfer.UpdateTransferRequest, diag.Diagnostics) {
+	r := new(transfer.UpdateTransferRequest)
+	var diags diag.Diagnostics = nil
+	var dgs diag.Diagnostics = nil
+
+	r.TransferId = m.Id.ValueString()
+	r.Name = m.Name.ValueString()
+	r.Description = m.Description.ValueString()
+
+	if m.Transformation != nil {
+		r.Transformation, dgs = m.Transformation.AsRequest(requestTypeUpdate)
+		diags.Append(dgs...)
+	}
+
+	return r, diags
+}
+
+func (m *transferResourceModel) DeleteRequest() (*transfer.DeleteTransferRequest, diag.Diagnostics) {
+	return &transfer.DeleteTransferRequest{
+		TransferId: m.Id.ValueString(),
+	}, nil
+}
+
+type transferTransformation struct {
+	Transformers []transferTransformer `tfsdk:"transformers"`
+}
+
+func transferTransformationSchema() schema.Attribute {
+	return schema.SingleNestedAttribute{
+		Attributes: map[string]schema.Attribute{
+			"transformers": schema.ListNestedAttribute{
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"replace_primary_key": transferTransformerReplacePrimaryKeySchema(),
+						"convert_to_string":   transferTransformerConvertToStringSchema(),
+						"dbt":                 transferTransformerDBTSchema(),
+						"table_splitter":      transferTransformerTableSplitterSchema(),
+					},
+				},
+				Optional: true,
+			},
+		},
+		Optional: true,
+	}
+}
+
+func (m *transferTransformation) AsRequest(rqt requestType) (*transfer.Transformation, diag.Diagnostics) {
+	r := new(transfer.Transformation)
+	var diags diag.Diagnostics = nil
+	var dgs diag.Diagnostics = nil
+
+	if len(m.Transformers) > 0 {
+		r.Transformers = make([]*transfer.Transformer, len(m.Transformers))
+		for i := range m.Transformers {
+			r.Transformers[i], dgs = m.Transformers[i].AsRequest(rqt)
+			diags.Append(dgs...)
+		}
+	}
+
+	return r, diags
+}
+
+func (m *transferTransformation) Read(t *transfer.Transformation) diag.Diagnostics {
+	var diags diag.Diagnostics = nil
+
+	tTransformers := t.GetTransformers()
+	if len(tTransformers) > 0 {
+		if len(m.Transformers) == 0 {
+			m.Transformers = make([]transferTransformer, 0)
+		}
+		for i := range tTransformers {
+			if i >= len(m.Transformers) {
+				m.Transformers = append(m.Transformers, *new(transferTransformer))
+			}
+			diags.Append(m.Transformers[i].Read(tTransformers[i])...)
+		}
+	} else {
+		m.Transformers = nil
+	}
+
+	return nil
+}
+
+type transferTransformer struct {
+	replacePrimaryKey *transferTransformerReplacePrimaryKey `tfsdk:"replace_primary_key"`
+	convertToString   *transferTransformerConvertToString   `tfsdk:"convert_to_string"`
+	dbt               *transferTransformerDBT               `tfsdk:"dbt"`
+	tableSplitter     *transferTransformerTableSplitter     `tfsdk:"table_splitter"`
```

Comment:
Currently, this does not work. When this structure is loaded (when e.g. a `Create` request with transformers filled is executed), Terraform returns an error message

```
    transfer_resource_test.go:18: Step 1/1 error: Error running apply: exit status 1
        
        Error: Value Conversion Error
        
          with doublecloud_transfer.ttr-transfer,
          on terraform_plugin_test.tf line 45, in resource "doublecloud_transfer" "ttr-transfer":
          45: 								transformers {
        
        An unexpected error was encountered trying to convert tftypes.Value into
        provider.transferTransformer. This is always an error in the provider. Please
        report the following to the provider developer:
        
        mismatch between struct and object: Object defines fields not found in
        struct: convert_to_string, dbt, replace_primary_key, and table_splitter.
```

I tried different approaches and checked the code of Terraform framework, but did not find a solution to this problem. It is persistent.

When exploring the internals of the framework, I found out that the tags are erased at the moment when this structure is filled (so there are no tags). Due to the absence of tags, the fields are not parsed as they should.